cdhweb/pages/views.py

```diff
@@ -50,3 +60,49 @@ def last_modified(self):
                 queryset.order_by(self.lastmodified_attr).reverse().first(),
                 self.lastmodified_attr,
             )
+
+
+class SiteSearchView(ListView, FormMixin):
+    """Search across all pages on the site."""
+
+    model = Page
+    form_class = SiteSearchForm
+    paginate_by = 10
+    page_title = "Search"
+    template_name = "cdhpages/search.html"
+
+    # NOTE filtering by page type currently not implemented in UI
+    filter_models = {
+        "everything": Page,
+        "people": Profile,
+        "updates": BlogPost,
+        "projects": Project,
+        "events": Event,
+    }
+
+    def get_queryset(self):
+        # NOTE not in UI — choose model to search across; Page for everything
+        # filter = self.request.GET.get("filter", "everything")
+        # model = self.filter_models[filter]
+        model = Page
```

Comment:
It still seems weird to me to set/override the model here when you already have it on the class.

When you add model filtering, does it _need_ to query based on the subclass, or can you something more like a normal queryset filter that looks for the related page type? That approach seems more flexible to me — and would make it possible to allow more than one kind of content.

Reply:
I hadn't thought of filtering based on related page type! The wagtail docs seem to prefer you to do as much as possible on the queryset (including switching the actual model) before calling `search()`, so I wasn't thinking about doing it that way...but I bet it'd work! will try

Reply:
reverted this to just use `self.model` for now since the filtering will happen as part of #360